openshift/edit_resources.rst

```diff
@@ -0,0 +1,126 @@
+Edit resources
+==============
+
+Deployment Config
+-----------------
+
+You can modify the config using ``oc edit dc ${INSTALLATION}``.
+
+Adjust Memory / CPU
+^^^^^^^^^^^^^^^^^^^
+
+You can change the resource for any of the containers. Here is how it looks like for Nice:
+
+.. code:: yaml
+
+   containers:
+   - name: nice
+     resources:
+       limits:
+         cpu: "1"
+         memory: 2Gi
+       requests:
+         cpu: 100m
+         memory: 1Gi
+
+The **requests** section describes the guaranteed resources and **limits** the maximum available. Both are optional. The
+OpenShift documentation has some more details on `requests and limits`_.
+
+.. _Requests and Limits: https://docs.openshift.org/latest/admin_guide/overcommit.html#requests-and-limits
+
+
+Setting Java and Nice Parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``env`` section can be used to change Java Paramaters, application.properties and hikaricp.properties.
+
+The env section looks something like this:
+
+.. code:: yaml
+
+    containers:
+      - name: nice
+        env:
+        - name: NICE2_HIKARI_dataSource__databaseName
+          value: nice_pege
+        - name: NICE2_HIKARI_dataSource__serverName
+          value: postgresqlssd
+        - name: NICE2_HIKARI_dataSource__user
+          value: nice_pege
+        - name: NICE2_JAVA_OPT____Dch__tocco__nice2__runenv
+          value: production
+        - name: NICE2_APP_nice2__enterprisesearch__solrUrl
+          value: http://solr-pege:8983/solr/nice2_index
+
+
+The following environment variables are supported:
+
+===================  ===================================================================================================
+NICE2_APP_*          Add custom entries to ``application.local.properties``.
+NICE2_HIKARI_*       Add custom entries to ``hikiricp.local.properties``.
+NICE2_JAVA_PARAM_*   Pass custom parameters to Java.
+NICE2_NICE_ARG_*     Pass custom argument to Nice.
+FLUENTD_TARGET_HOST  Target host for fluentd. (Fluentd forwarding is enabled automatically if this variable is set)
+FLUENTD_TARGET_PORT  Target port for fluentd. (defaults to 24224)
+===================  ===================================================================================================
+
+.. important::
+    Openshift does not currently allow '.' or '-' to appear as key of an environment variable. [#f2]_
```

Comment:
Don't think it really matters. Do you? If "keys" it used should also be "variables" I guess.